### PR TITLE
test: set up e2e test pipeline on ADO

### LINKED ISSUE #568: Move e2e test pipeline from Jenkins to ADO
**Describe the bug**

**Steps To Reproduce**

**Expected behavior**

**AAD Pod Identity version**

**Kubernetes version**

**Additional context**

### PR DESCRIPTION
**Pull request by chewong**
**Reason for Change**:

Adding e2e tests on ADO as a PR gate to reduce regressions introduced by PRs.

**Issue Fixed**:

Fixes #568 

**Notes for Reviewers**:

### CODE REVIEW

**Comment by chewong** [comment]
I decided to run a limited amount of test cases listed below for PR gates to reduce test duration and flakiness (I noticed that running on a fresh cluster is much flakier than running in Jenkins). I hope we can increase the number of test cases and reduce test duration + flakiness by fixing https://github.com/Azure/aad-pod-identity/issues/586.

- should pass the identity validating test
- should not pass the identity validating test if the AzureIdentity is deleted
- should not pass the identity validating test if the AzureIdentityBinding is deleted
- should delete the AzureAssignedIdentity if the deployment is deleted
- should cleanup iptable rules after deleting aad-pod-identity
- should not delete the Immutable Identity from vmss when the deployment is deleted
- should pass liveness probe test
should delete assigned identity when identity no longer exists on underlying node
- should pass multiple identity validating test even when MIC is failing over
- should pass multiple identity validating test even when MIC is failing over
- should assign new identity and remove old when AzureIdentity updated

File: .pipelines/e2e-tests-template.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+              # Sleep for 120 seconds to wait for nodes and pods to become ready
+              sleep 120
+              kubectl wait --for=condition=ready node --all
+              kubectl wait pod -n kube-system --for=condition=Ready --all
+              kubectl get nodes -owide
+              kubectl cluster-info
+
+              # Install gatekeeper
+              kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/master/deploy/gatekeeper.yaml
```
Comment (aramase) [inline .pipelines/e2e-tests-template.yml:99]: Why are we installing gatekeeper?

File: test/e2e/cluster_configs/vmas.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "apiVersion": "vlabs",
+    "properties": {
+        "orchestratorProfile": {
+            "orchestratorType": "Kubernetes",
+            "orchestratorRelease": "$K8S_RELEASE"
+        },
```
Comment (aramase) [inline test/e2e/cluster_configs/vmas.json:7]: Let's also add the addon `aad-pod-identity` and set to false explicitly in the api model.

File: .pipelines/e2e-tests-template.yml (hunk 1 above)
Comment (chewong) [reply .pipelines/e2e-tests-template.yml]: Didn't realize we install gatekeeper during the test run

**Pull request merged**